src_py/cursors.py

```diff
@@ -119,15 +119,15 @@ def __hash__(self):
     def __repr__(self):
         if self.type == "system":
             id_string = _cursor_id_table.get(self.data[0], "constant lookup error")
-            return "<Cursor(type: system, constant: " + id_string + ")>"
+            return f"<Cursor(type: system, constant: {id_string})>"
         if self.type == "bitmap":
-            size = "size: " + str(self.data[0])
-            hotspot = "hotspot: " + str(self.data[1])
-            return "<Cursor(type: bitmap, " + size + ", " + hotspot + ")>"
+            size = f"size: {self.data[0]!s}"  # !s for calling str(), or it will call format()
```

Comment:
format? I think you mean str() vs repr().

I've never heard of this "!s" syntax, and it was difficult to Google. I finally found a reference in https://peps.python.org/pep-0498/#s-r-and-a-are-redundant, which calls them redundant. I would suggest using `str()` explicitly instead, that way is more easily understood, and doesn't need the comment at least.

What goes wrong with repr vs str here anyways, I'm having a hard time thinking it through

Reply:
I just changed the look of syntax, nothing changed, if I called f"{self.data}" without !s or str() it would call format() method instead of str() which was originally here. I can change it to str() method.

Reply:
>  if I called f"{self.data}" without !s or str() it would call format() method instead of str() which was originally here.

It would not call `format()`

Reply:
https://stackoverflow.com/a/72449614/18343885

Reply:
TIL there's a `__format__` dunder method.

Still, does it lead to different results?

Reply:
It probably won't because this variable contains only tuple[int] tuple tuple[, ...]

Reply:
That's why I was confused over your changes. I assumed you had tested this and it needed to be this way.

Reply:
I didn't want to change the behavior of function, but it seems there is no point in using that !s method